### DISCUSSION TITLE
Unable to disable writing DISCOGS_RELEASE_ID

### DISCUSSION THREAD

**Discussion by armak** [ghDaYuYu/foo_discogs#121] [Feature Request]
The option to disable writing `DISCOGS_RELEASE_ID` is either disabled, or does not work. Normally for some reason clicking on the right where the action column is, these options are disabled for a few of the ID tags. However, they can still be disabled for most of them by clicking on the left side. Even then however, the state doesn't properly disable for `DISCOGS_RELEASE_ID`. I do not want to write any of these ID tags into my files.

![image](https://github.com/user-attachments/assets/b506b808-0619-4074-8b44-89694d75cf74)

Note how the "write" action still remains enabled for the release ID.

**Comment by armak** [comment]
I was able work around this by manually this by exporting my config, editing the "write" field to "0", and reimporting, but doing so disabled the button to write the tags in the Preview window for some reason?

![image](https://github.com/user-attachments/assets/2dc1dc8c-36ad-4dbc-a86f-c68b65d7b24b)

The button in the main Track Matching dialog still works though, so I'm ok with it for now, but probably still worth fixing the ability to disable writing these IDs so that the user interface is consistent and there are no quirks when the user chooses not to have them written.

**Comment by ghuDaYuYu** [comment]
The user interface is quite consistent... alright not 100% consistent.
Grayed-out items usually mean 'disabled' not 'quirky item'.

The disabled 'Write tags' in the Preview means 'nothing to do'. That state is usually unknown in the previous dialog, so probably not related to your hack.
I see... unless coming back from the preview panel.

About the DISCOGS_RELEASE_ID...
The 'I do not want to write any of these ID tags into my files' is a fair argument and will be taken into consideration.
This issue will be closed soon. Might move it to the discussion area, as a feature request, idk yet.
Thanks for your feedback.

**Comment by frogworth** [comment]
@armak I think @ghuDaYuYu is working on a different workflow for this, but tags like that which are greyed out can still be changed by holding down shift when you click.

**Comment by ghuDaYuYu** [comment]
Hi [frogworth](https://github.com/frogworth),
You are right again! Let´s move the issue to the discussion area...

**Comment by ghuDaYuYu** [comment]
... I mentioned data consistency at HA. But why restrict just RELEASE_ID and not also MASTER_ID, LABEL_ID and ARTIST_ID ?
I believe there were also considerations related to degrading the UI. And probably, network traffic was also considered, maybe more than fifteen years ago.

**Comment by bassstorm** [comment]
I know how you feel @armak, same here :) The solution would probably be to just remove that entry from tag mappings altogether, but I never tried tbh.

**Comment by ghuDaYuYu** [comment]
You may consider another option.
'Plain write tags' features 'disable writing DISCOGS_RELEASE_ID'.
It is planned to remove the Find Releases window and make the option exclusive to betas.

This is the use case so far...
- Use WWW to save and restore release IDs originating from Discogs.com look ups or other sources.

**Comment by ghuDaYuYu** [comment]
Here is a new build.
- Simplified 'Plain write tags' entry form.
- Ignore auto-load preference options opening entry form.
- Cache: fix 'Delete artist cache' incomplete memory cache release removal.

[Discogger v1.0.21.2 Beta 4](https://gist.github.com/ghuDaYuYu/807a697f20b9a7cd3f71c2a3e21aa531/raw/35d151b2f8a2c04182d6b0702f5f51183f98145f/foo_discogger%25201.0.21.2%2520Beta%25204.zip)

**Comment by bassstorm** [comment]
Installed Beta 4 and tried "Plain write tags" flow (for the 1st time!). Following summary is not a negative note, but just my personal experience :)
Frankly, not sure how that helps with skipping DISCOGS_RELEASE_ID to begin with, I haven't found that disable option anywhere. But what's more important I don't see myself using that flow anyway, because I don't usually have release id/url handy.

**Comment by ghuDaYuYu** [comment]
The option should be in the tag mappings. Press shift + right click on DISCOGS_RELEASE_ID.
About workflows, no worries, no solution will fit all : )

**Reply by bassstorm** [reply]
> The option should be in the tag mappings. Press shift + right click on DISCOGS_RELEASE_ID. About workflows, no worries, no solution will fit all : )

Ah, that option :smile: Yes, that works :+1:

**Comment by ghuDaYuYu** [comment]
Yes, the 'secret' key 😄 
BTW, I forgot to link the [Discogs Tagger Use Case](https://github.com/ghDaYuYu/foo_discogger/blob/master/README.md), it might help contextualizing the topic.
